packages/lego-bricks/src/components/Modal/Modal.tsx

```diff
@@ -1,85 +1,92 @@
+ *
+ * ### Example Usage (controlled)
+ * ```tsx
+ * const [isOpen, setIsOpen] = useState(false);
+ * return (
+ *   <>
+ *     <Button onPress={() => setIsOpen(true)}>Open modal</Button>
+ *     <Modal
+ *       isOpen={isOpen}
+ *       onOpenChange={setIsOpen}
+ *       title="Modal title"
+ *     >
+ *       {modal content}
+ *     </Modal>
+ *   </>
+ * );
+ * ```
+ *
+ * ### Example Usage (uncontrolled)
+ * ```tsx
+ * <DialogTrigger>
+ *   <Button>Open modal</Button>
+ *   <Modal title="Modal title">
+ *     {modal content}
+ *   </Modal>
+ * </DialogTrigger>
+ * ```
  */
 const Modal = ({
-  show,
-  children,
-  onHide,
-  backdrop = true,
-  closeOnBackdropClick = true,
-  keyboard = true,
+  isOpen,
+  onOpenChange,
+  isDismissable = true,
   contentClassName,
-  backdropClassName,
+  dialogRole,
+  title,
+  children,
 }: Props) => {
-  const handleBackdropClick = () => {
-    if (closeOnBackdropClick) {
-      onHide();
-    }
-  };
-
-  const handleBackdropKeyDown = (event: KeyboardEvent<HTMLDivElement>) => {
-    if (
-      closeOnBackdropClick &&
-      (event.key === 'Enter' || event.key === 'Escape' || event.key === ' ')
-    ) {
-      event.preventDefault();
-      onHide();
-    }
-  };
-
   return (
-    <ReactModal
-      className={cx(styles.content, contentClassName, {
-        [styles.show]: show,
-      })}
-      show={show}
-      backdrop={backdrop}
-      onHide={onHide}
-      keyboard={keyboard}
-      renderBackdrop={(props) => (
-        <div
-          {...props}
-          className={cx(backdropClassName || styles.backdrop, {
-            [styles.show]: show,
-          })}
-          onClick={handleBackdropClick}
-          onKeyDown={handleBackdropKeyDown}
-          tabIndex={closeOnBackdropClick ? 0 : undefined} // Make it focusable if clickable
-          role={closeOnBackdropClick ? 'button' : undefined}
-        />
-      )}
-      data-test-id="Modal__content"
+    <ModalOverlay
+      isOpen={isOpen}
+      onOpenChange={onOpenChange}
+      isDismissable={isDismissable}
+      isKeyboardDismissDisabled={!isDismissable}
+      className={styles.overlay}
     >
-      <>
-        <Icon
-          name="close"
-          onClick={onHide}
-          className={styles.closeButton}
-          data-test-id="Modal__closeButton"
-        />
-        {children}
-      </>
-    </ReactModal>
+      <AriaModal className={cx(styles.modal, contentClassName)}>
+        <Dialog role={dialogRole} data-test-id="Modal__content">
+          {({ close }) => (
+            <>
+              {title && (
+                <Heading slot="title" className={styles.title}>
+                  {title}
+                </Heading>
+              )}
+              <Icon
+                name="close"
+                onClick={close}
+                className={styles.closeButton}
+                data-test-id="Modal__closeButton"
+              />
+              {typeof children === 'function' ? children({ close }) : children}
```

Comment:
Care to explain? 😬

Reply:
Sure, this is for cases where you want to add something within the modal that closes it. In that case you would do something like this:
```tsx
<Modal>
	{({ close }) => (
		{...modal content}
		<Button onPress={close}>Close the modal</Button>
	)}
</Modal>
```
Which is how it's done in the default `react-aria-components`-modal. But we also want to support just having normal react-nodes as children, so we do this check and handle the children in the appropriate way.

Reply:
ah thanks yes that makes perfect sense! I don't know what confused me anymore